obspy/core/tests/test_waveform_plugins.py

```diff
@@ -155,6 +162,18 @@ def test_isFormat(self):
         ]
         formats_ep = _get_entry_points('obspy.plugin.waveform', 'isFormat')
         formats = list(formats_ep.values())
+        # Get all the test directories.
+        paths = {}
+        all_paths = []
+        for f in formats:
+            sep = os.path.sep
+            path = f.dist.location + sep + f.module_name.replace('.', sep)
+            path = path.rsplit(sep, 1)[0] + sep + 'tests' + sep + 'data'
```

Comment:
Assuming we want to merge this: Can you replace the previous couple of lines and use `os.path.join`, `os.path.dirname`, and consorts? That should be a bit safer for funny path names and other issues.